client/landing/stepper/declarative-flow/internals/components/redirect-to-step/index.tsx

```diff
@@ -0,0 +1,21 @@
+import { type FC } from 'react';
+import { generatePath, Navigate, useParams } from 'react-router';
+import { useFlowLocale } from 'calypso/landing/stepper/hooks/use-flow-locale';
+import { type StepperStep } from '../../types';
+
+interface Props {
+	slug: StepperStep[ 'slug' ];
+}
+
+export const RedirectToStep: FC< Props > = ( { slug } ) => {
+	const lang = useFlowLocale();
+	const { flow } = useParams();
+
+	const to = generatePath( '/:flow/:step/:lang?', {
+		flow: flow!,
+		step: slug,
+		lang: lang === 'en' ? null : lang,
+	} );
```

Comment:
I think we should only add the `lang` param if the user is logged out. It doesn't do anything for logged-in users.

Reply:
I think it makes sense, I implemented it and add some automated tests  [052175e](https://github.com/Automattic/wp-calypso/pull/91935/commits/052175ee334c7ec28e1dc41f9a1a453f04202acc)